mslib/msui/kmloverlay_dockwidget.py

```diff
@@ -290,10 +290,13 @@ def __init__(self, parent=None, view=None):
         self.pushButton_unselect_all.clicked.connect(self.unselect_all)
         self.pushButton_merge.clicked.connect(self.merge_file)
         self.labelStatusBar.setText("Status: Click on Add KML Files to get started.")
+        self.dsbx_linewidth.setSpecialValueText("None")  # None is the value of double spin box when it is minimum i.e 0
+        self.dsbx_linewidth.setValue(0.0)  # sets initial value as 0 for linewidth
 
-        self.dialog = CustomizeKMLWidget(self)  # create object of dialog UI Box
-        self.listWidget.itemDoubleClicked.connect(self.open_customize_kml_dialog)
-        self.dialog.pushButton_colour.clicked.connect(self.select_color)
+        # Connect slots and signals (For showing color and linewidth of clicked file).
+        self.listWidget.itemClicked.connect(self.show_linewidth)
+        self.pushButton_color.clicked.connect(self.select_color)
+        self.dsbx_linewidth.valueChanged.connect(self.select_linewidth)
 
         self.listWidget.itemChanged.connect(self.load_file)  # list of files in ListWidget
```

Comment:
This triggers also when you change the icons for the items.
So whenever you change the colour or linewidth, self.load_file is executed twice in the process which is very slow.
This should first be disconnected before doing so.

There may be more such traps you ran into which cause the whole thing to slow down significantly.
Check for any redundant method calls.

Reply:
Okay @Marilyth I would try to work on the points that you mentioned.

Reply:
Thanks for the review.